client/src/components/RequestAccessButton.tsx

```diff
@@ -0,0 +1,30 @@
+"use client";
+import { useState } from "react";
+
+const RequestAccessButton = () => {
+  const [accessRequested, setAccessRequested] = useState(false);
+
+  return (
+    <div
+      style={{
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "center",
+        padding: "10px 20px",
+        fontSize: "16px",
+        borderRadius: "5px",
+        border: `2px solid ${
+          accessRequested ? "rgba(255, 255, 255, 0.5)" : ""
+        }`,
+        color: accessRequested ? "rgba(255, 255, 255, 0.5)" : "",
+        cursor: accessRequested ? "default" : "pointer",
+        WebkitTapHighlightColor: "transparent",
+      }}
+      onClick={accessRequested ? undefined : () => setAccessRequested(true)}
```

Comment:
Instead of setting all these manually, I think it would make more sense to make this a button and do `disabled={accessRequested}`. If the default styling change isn't quite right, you can use the CSS `disabled` selector